src/main/java/com/example/mybookshopapp2/controllers/GenresController.java

```diff
@@ -1,29 +1,36 @@
 package com.example.mybookshopapp2.controllers;
 
-import com.example.mybookshopapp2.data.Author;
-import com.example.mybookshopapp2.data.AuthorService;
+import com.example.mybookshopapp2.data.SearchWordDto;
+import com.example.mybookshopapp2.model.Book;
+import com.example.mybookshopapp2.model.Genre;
+import com.example.mybookshopapp2.service.GenreService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.ModelAttribute;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 @Controller
 public class GenresController {
```

Comment:
I'm not sure this is valid controller. RequestMapping is missing

Reply:
```
@GetMapping("/genres")
    public String genresPage() {
        logger.debug("Rendering genres/index.html");
        return "/genres/index";
    }
```